tests/unit/plugins/inventory/test_kubevirt.py

```diff
@@ -1143,3 +1145,65 @@ def test_set_ansible_host_and_port(monkeypatch, request, inventory, host_vars, v
     inventory.set_ansible_host_and_port(Dict(vmi), vmi_name, "10.10.10.10", service, opts)
 
     assert request.getfixturevalue(result) == host_vars
+
+
+@pytest.fixture(scope="function")
+def body_error(mocker):
+    error = DynamicApiError(e=mocker.Mock())
+    error.headers = None
+
+    body = "This is a test error"
+    error.body = body
+
+    return error
+
+
+@pytest.fixture(scope="function")
+def message_error(mocker):
+    error = DynamicApiError(e=mocker.Mock())
+    error.headers = {
+        "Content-Type": "application/json"
+    }
+
+    body = {
+        "message": "This is a test error"
+    }
+
+    json_data = json.dumps(body)
+    byte_array = json_data.encode('utf-8')
+    error.body = byte_array
+
+    return error
+
+
+@pytest.fixture(scope="module")
+def message_error_return():
+    return "This is a test error"
+
+
+@pytest.fixture(scope="function")
+def status_reason_error(mocker):
+    error = DynamicApiError(e=mocker.Mock())
+    error.body = None
+    error.status = 404
+    error.reason = "This is a test error"
+    return error
+
+
+@pytest.fixture(scope="module")
+def status_reason_error_return():
```

Comment:
Do we really need separate fixtures for simple strings? I'd just define the in the parametrize block below.

Reply:
You are right! I've fallen into the bad habit of defining everything as fixtures. 
Fixed.